examples/window/window_settings.rs

```diff
@@ -10,12 +10,25 @@ fn main() {
             vsync: true,
             ..Default::default()
         })
+        .insert_resource(IconResource::default())
         .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
         .add_system(change_title.system())
         .add_system(toggle_cursor.system())
+        .add_system(toggle_icon.system())
         .run();
 }
 
+#[derive(Debug, Clone, Default)]
+struct IconResource {
+    handle: Handle<Texture>,
+}
+
+fn setup(asset_server: Res<AssetServer>, mut icon_resource: ResMut<IconResource>) {
+    let icon = asset_server.load("android-res/mipmap-mdpi/ic_launcher.png");
+    (*icon_resource).handle = icon;
```

Comment:
```suggestion
    icon_resource.handle = icon;
```

My intuition is that this should work due to Rust's automatic dereferencing.